controllers/imageclusterinstall_controller.go

```diff
@@ -697,33 +697,27 @@ func (r *ImageClusterInstallReconciler) getBMH(ctx context.Context, bmhRef *v1al
 	return bmh, nil
 }
 
-func (r *ImageClusterInstallReconciler) removeBMHDataImage(ctx context.Context, log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, bmhRef *v1alpha1.BareMetalHostReference) error {
-	dataImage := &bmh_v1alpha1.DataImage{}
-
-	key := types.NamespacedName{
-		Name:      bmhRef.Name,
-		Namespace: bmhRef.Namespace,
+func (r *ImageClusterInstallReconciler) removeBMHDataImage(ctx context.Context, log logrus.FieldLogger, bmhRef types.NamespacedName) (*bmh_v1alpha1.DataImage, error) {
+	dataImage, err := r.deleteDataImage(ctx, log, bmhRef)
+	if err != nil || dataImage == nil {
```

Comment:
If the data image was already removed (`err == nil && dataImage == nil`) should we still attach and reboot the BMH? Looks like we're exiting early in that case here.

Reply:
The only reason we attach and reboot is to allow the dataImage to be deleted.
If the dataImage doesn't exist we don't want to touch the BMH.

Reply:
Ah okay